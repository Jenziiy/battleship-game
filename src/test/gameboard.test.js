const GameBoard = require('../gameboard.js');
const Ship = require('../ship.js');

test('gameboard loads fleet', () => {
  expect(new GameBoard()).toMatchSnapshot({"ships":[
   { length: 4, coords: null, hits: 0, sunk: false },
   { length: 3, coords: null, hits: 0, sunk: false },
   { length: 3, coords: null, hits: 0, sunk: false },
   { length: 2, coords: null, hits: 0, sunk: false },
   { length: 2, coords: null, hits: 0, sunk: false },
   { length: 2, coords: null, hits: 0, sunk: false },
   { length: 1, coords: null, hits: 0, sunk: false },
   { length: 1, coords: null, hits: 0, sunk: false },
   { length: 1, coords: null, hits: 0, sunk: false },
   { length: 1, coords: null, hits: 0, sunk: false }]})
})