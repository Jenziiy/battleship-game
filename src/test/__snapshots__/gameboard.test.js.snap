// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gameboard loads fleet 1`] = `
{
  "ships": [
    {
      "coords": null,
      "hits": 0,
      "length": 4,
      "sunk": false,
    },
    {
      "coords": null,
      "hits": 0,
      "length": 3,
      "sunk": false,
    },
    {
      "coords": null,
      "hits": 0,
      "length": 3,
      "sunk": false,
    },
    {
      "coords": null,
      "hits": 0,
      "length": 2,
      "sunk": false,
    },
    {
      "coords": null,
      "hits": 0,
      "length": 2,
      "sunk": false,
    },
    {
      "coords": null,
      "hits": 0,
      "length": 2,
      "sunk": false,
    },
    {
      "coords": null,
      "hits": 0,
      "length": 1,
      "sunk": false,
    },
    {
      "coords": null,
      "hits": 0,
      "length": 1,
      "sunk": false,
    },
    {
      "coords": null,
      "hits": 0,
      "length": 1,
      "sunk": false,
    },
    {
      "coords": null,
      "hits": 0,
      "length": 1,
      "sunk": false,
    },
  ],
}
`;
